### PR TITLE
Implement exporter retry logic

### LINKED ISSUE #31: Implement retry logic in exporter
Currently messages are dropped in exporter if exporting fails. Implement retry logic that retries messages and has the following features:

- Exponential backoff
- Random jitter

For now, we can just buffer unsent messages in crossbeam channel (don't pull new message before previous one is sent). Later we can think about implementing some more sophisticated caching mechanism.

### PR DESCRIPTION
**Pull request by ollipa**
Closes #31

### CODE REVIEW

**Comment by ollipa** [comment]
I added `max_backoff` value that is now hardcoded to 2 minutes and modified the implementation so that instead of starting to export at full speed after first successful export, backoff is decreased on every successful delivery until it reaches to zero.

I also did some refactoring to split the logic in `ExportScheduler` to multiple smaller functions. I removed `register` method from `ExportScheduler` as well, because I'm not sure if/when we will support configuring multiple exporters so maybe it is better not to make changes for that in advance.

File: src/exporters/mod.rs [hunk 1]
```diff
@@ -5,54 +5,137 @@ mod stdout;
...
 
+extern crate rand;
 use crossbeam_channel::Receiver;
+use rand::Rng;
 
 use crate::message::Message;
 pub use influxdb::InfluxDB;
 pub use stdout::Stdout;
 
+const ZERO: Duration = Duration::from_secs(0);
```
[the remark this answers is not in the document]
Comment (ollipa) [reply src/exporters/mod.rs:16]: I will make this change in PR #38

File: src/exporters/mod.rs [hunk 2]
```diff
@@ -5,54 +5,142 @@ mod stdout;
...
-                    thread::sleep(Duration::from_secs(1));
-                }
+            self.export(message);
+            self.backoff_sleep();
+        }
+
+        println!("Collectors disconnected. Stopping exporter.");
+    }
+
+    fn receive(&mut self) -> Option<Message> {
```
Comment (adarshk7) [inline src/exporters/mod.rs:52]: Return `Result` here instead of `Option`

Comment (ollipa) [reply src/exporters/mod.rs:52]: I will make this change in PR #38

**Pull request merged**